greenheart/simulation/greenheart_simulation.py

```diff
@@ -210,133 +434,448 @@ def simple_solver(initial_guess=0.0):
+        )
+        lcoh, pf_lcoh = he_fin.run_profast_full_plant_model(
+            greenheart_config,
+            orbit_config,
+            orbit_project,
+            electrolyzer_physics_results,
+            capex_breakdown,
+            opex_breakdown_annual,
+            hopp_results,
+            incentive_option,
+            design_scenario,
+            total_accessory_power_renewable_kw,
+            total_accessory_power_grid_kw,
+            verbose=verbose,
+            show_plots=show_plots,
+            save_plots=save_plots,
+        )
+
+        if "steel" in greenheart_config:
+            if verbose:
+                print("Running steel\n")
+
+            steel_costs = greenheart_config["steel"]["costs"]
+            steel_capacity = greenheart_config["steel"]["capacity"]
+            feedstocks = SteelFeedstocks(**steel_costs["feedstocks"])
+
+            # run steel capacity model to get steel plant size
+            # uses hydrogen amount from electrolyzer physics model
+            capacity_config = SteelCapacityModelConfig(
+                feedstocks=feedstocks,
+                hydrogen_amount_kgpy=electrolyzer_physics_results["H2_Results"][
+                    "hydrogen_annual_output"
+                ],
+                **steel_capacity
+            )
+            capacity = run_size_steel_plant_capacity(capacity_config)
+
+            # run steel cost model
+            # uses plant capacity from steel capacity model
+            # uses lcoh from profast
+            steel_costs["feedstocks"] = feedstocks
+            steel_cost_config = SteelCostModelConfig(
+                plant_capacity_mtpy=capacity.steel_plant_capacity_mtpy,
+                lcoh=lcoh,
+                **steel_costs
+            )
+            steel_cost_config.plant_capacity_mtpy = capacity.steel_plant_capacity_mtpy
+            steel_costs = run_steel_cost_model(steel_cost_config)
+
+            # run steel finance model
+            # uses plant capacity from steel capacity model
+            # uses lcoh from profast
+            # uses costs from steel cost model
+            # uses feedstocks from steel cost model
+            steel_finance = greenheart_config["steel"]["finances"]
+            steel_finance["feedstocks"] = feedstocks
+
+            steel_finance_config = SteelFinanceModelConfig(
+                plant_capacity_mtpy=capacity.steel_plant_capacity_mtpy,
+                plant_capacity_factor=capacity_config.input_capacity_factor_estimate,
+                steel_production_mtpy=run_steel_model(
+                    capacity.steel_plant_capacity_mtpy,
+                    capacity_config.input_capacity_factor_estimate,
+                ),
+                costs=steel_costs,
+                lcoh=lcoh,
+                **steel_finance
+            )
+            steel_finance = run_steel_finance_model(steel_finance_config)
+
+        if "ammonia" in greenheart_config:
+            if verbose:
```

Comment:
Same comment as for steel

Reply:
I wonder about this changing the config dictionary. That feels a little dangerous. Let's discuss this in our code meeting.

Reply:
I agree. As it currently stands, this will overwrite some configs even if the user does not want that. I was thinking one option is to first check if the necessary field is user-provided, and calculate if not

Reply:
I think it may be best to create an output data class where outputs can be stored. We should probably have a separate entry in the Greensteel config for steel and ammonia to specify if  the user provided LCOH or the calculated LCOH should be used so the knock-on effects of providing an LCOH at input are clear.